troubleshooting-guide/action-errors.md

```diff
@@ -1,15 +1,331 @@
+
+``` 
+Parameter 'Expiry Duration of Signed URL' is NOT a number
+```
+
+This message indicates that the query parameter mentioned in the message expects a number but a non-numerical value has 
+been provided in the [query form](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connecting-to-databases/querying-a-database). 
+
+This error can be fixed by editing the [query form](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connecting-to-databases/querying-a-database) and providing a valid number as input for the mentioned parameter.
+
+### JSON Parsing Error
+
+``` 
+Error parsing the JSON body
+```
+
+``` 
+Error converting array to ND-JSON
+```
+
+``` 
+Unable to parse condition value as a JSON list
+```
+
+This message indicates that the [JSON](https://www.w3schools.com/whatis/whatis_json.asp#:~:text=JSON%20stands%20for%20JavaScript%20Object,describing%22%20and%20easy%20to%20understand) string passed to 
+the query as a parameter is not a valid JSON string.
+
+This error can be fixed by editing the [query form](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connecting-to-databases/querying-a-database) and passing a valid JSON string as parameter.
+
+## REST API Errors
+
+![Click to expand](../.gitbook/assets/api-error.png)
+
+### Missing URL Error
+
+```
+DEFAULT_REST_DATASOURCE is not correctly configured. Please fix the following and then re-run: \n[Missing URL.]
+```
+
+This message indicates that the REST API's URL field in the [API editor form](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connect-to-apis/execute-apis) has been left empty. 
+
+This error can be fixed by editing the [REST API form](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connect-to-apis/execute-apis) and providing a URL. 
+    
+### Missing Client Secret / Client ID / Access Token Error
+
+``` 
+DEFAULT_REST_DATASOURCE is not correctly configured. Please fix the following and then re-run: \n[Missing Client Secret, Missing Client ID, Missing Access Token URL]
+```
+
+This message indicates that the mentioned parameter fields - `Client Secret` / `Client ID` / `Access Token URL` have 
+been left empty. These fields are nested in the `Authentication` sub-section which becomes visible if the 
+`Authentication Type` field has been chosen as [OAuth 2.0](https://docs.appsmith.com/core-concepts/connecting-to-data-sources/connect-to-apis/authentication/oauth2-authentication)
+
+### Secret Key Required Error
+
+``` 
+Secret key is required when sending session details is switched on, and should be at least 32 characters in length.
+```
+
+This message indicates that `Send Appsmith signature header` field has been marked as `Yes` but the `Session Details Signature Key` field is left empty.
+
+This error can be resolved by filling in the `Session Details Signature Key` field or by disabling the  `Send Appsmith signature header` field by selecting `No`.
+
+## Page Access Error
```

Comment:
I think this can go under the Application Errors section as another sub-section.

Reply:
Done.